dialect/sql/schema/inspect.go

```diff
@@ -62,6 +62,24 @@ func (i *Inspector) Tables(ctx context.Context) ([]*Table, error) {
 		}
 		tables = append(tables, t)
 	}
+
+	fki, ok := i.sqlDialect.(interface {
+		foreignKeys(context.Context, dialect.Tx, []*Table) (map[*Table][]*ForeignKey, error)
+	})
+	if ok {
+		allFks, err := fki.foreignKeys(ctx, tx, tables)
+		for _, t := range tables {
+			if tableFks, ok := allFks[t]; ok {
+				for _, fk := range tableFks {
+					t.AddForeignKey(fk)
+				}
+			}
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return tables, nil
```

Comment:
```suggestion
	if ok {
		if err := fki.foreignKeys(ctx, tx, tables); err != nil {
			return nil, err
		}
	}
	return tables, nil
```

See my comment above.